spring-batch-infrastructure/src/test/java/org/springframework/batch/item/jms/builder/JmsItemReaderBuilderTests.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.batch.item.jms.builder;
+
+import java.util.Date;
+
+import javax.jms.Message;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.springframework.batch.item.jms.JmsItemReader;
+import org.springframework.jms.core.JmsOperations;
+import org.springframework.jms.core.JmsTemplate;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Glenn Renfro
+ */
+public class JmsItemReaderBuilderTests {
+
+	private JmsOperations defaultJmsTemplate;
+
+	@Before
+	public void setupJmsTemplate() {
+		this.defaultJmsTemplate = mock(JmsOperations.class);
+		when(this.defaultJmsTemplate.receiveAndConvert()).thenReturn("foo");
+	}
+
+	@Test
+	public void testNoItemType() {
+		JmsItemReader<String> itemReader = new JmsItemReaderBuilder<String>().jmsTemplate(this.defaultJmsTemplate)
+				.build();
+		assertEquals("foo", itemReader.read());
+	}
+
+	@Test
+	public void testSetItemType() {
```

Comment:
Does this test actually validate the condition?  Since the behavior is the same as the previous test, I'd assume that you'd need to set a type other than String to validate that this worked...